### PR TITLE
Menu: Fix enter and spacebar behaviour, focus by first letter, close nested submenus

### PR DESCRIPTION
**Pull request by juleskuehn**
Thoroughly tested in Chrome 67, Firefox 61, IE11.
Fixes Issue #8418 and a number of undocumented issues.

Fixes to behaviour:
- Spacebar and enter now behave as in ARIA recommendation (except in IE11 detail/summary).
 - As with right arrow key, enter or spacebar expands a submenu and focuses its first item.
 - If item has no submenu (ie. is a link), activates the item by emulating click.
- Nested submenus close on blur (except in mobile menu).
- Nested submenus and their items can be selected by letter from a focused top-level menuitem.

Code style improvements:
- symbolic variables for keycodes
- remove unneeded conditionals

Unresolved issues:
- Mobile menus do not collapse on blur.
- IE11 does not allow enter and spacebar to expand detail/summary with correct behaviour.

### CODE REVIEW

**Review by duboisp — CHANGES_REQUESTED** [review]
@juleskuehn, your PR seems to include unrelated files change, I will help you to cleaning up at the next code sprint.

**Comment by juleskuehn** [comment]
I was unable to resolve the issue in IE11 without putting a check for role="menuitem" in the details.js polyfill. Looking forward to discussing this with you at the code sprint.

**Comment by juleskuehn** [comment]
I was able to resolve the issue in IE11 by putting a check for role "menuitem" in the details.js polyfill.

`else if ( ( which === 13 || which === 32 ) && currentTarget.getAttribute( "role" ) !== "menuitem" )`

Since the polyfill is used in other plugins, I tested the other working examples with my changes in place:

- [x] Charts (all)
- [x] Collapsible alerts
- [x] Details/summary polyfill
- [x] Tabbed interface
- [x] Tabbed Interface - Carousel
- [x] Toggle
- [x] Multimedia Player - Video

File: src/polyfills/details/details.js [hunk 1]
```diff
@@ -107,7 +107,9 @@ $document.on( "click keydown toggle." + componentName, selector, function( event
 		( event.type === "toggle" && event.namespace === componentName ) ) ) {
 		details = currentTarget.parentNode;
 		details.open = !details.open;
-	} else if ( which === 13 || which === 32 ) {
+
+	// Prevent double handling of enter / spacebar if within a menu (handled by menu.js)
+	} else if ( ( which === 13 || which === 32 ) && currentTarget.getAttribute( "role" ) !== "menuitem" ) {
```
Comment (nschonni) [inline src/polyfills/details/details.js]: Is there another way to handle this? The polyfill shouldn't really care about other WET plug-ins

Comment (juleskuehn) [reply src/polyfills/details/details.js]: I was able to fix it by adding `event.stopImmediatePropagation()` in menu.js. This depends on the menu.js handler being called before the details.js handler, which it is. Is this an acceptable solution?

**Review by duboisp — APPROVED** [review]

**Pull request merged**